Tasks/SonarQubePostTest/PRCA/PostComments-Server.ps1

```diff
@@ -327,4 +283,53 @@ function GetCodeFlowChangeTrackingId
     return $change.ChangeTrackingId
 } 
 
-#endregion 
\ No newline at end of file
+#endregion 
+
+#
+# Invoke a method with optional params by reflection. 
+# This can be used to overcome PS bugs in determining the correct overload candidate method from a .net assembly
+#
+function InvokeByReflection
+{
+    param ($obj, $methodName, [Type[]] $parameterTypes, [Object[]] $parameterValues)
+
+    $publicMethods = $obj.GetType().GetMethods() | Where-Object {($_.Name -eq $methodName) -and  ($_.IsPublic -eq $true)}
+    Assert ($publicMethods.Count -gt 0) "$methodName not found"
+
+    foreach ($method in $publicMethods)
+    {
+        $methodParams = $method.GetParameters();
+        if ((ParamTypesMatch $methodParams $parameterTypes) -eq $true) 
+        {
+            $paramValuesAndDefaults = New-Object "System.Collections.Generic.List[Object]"
+            $paramValuesAndDefaults.AddRange($parameterValues);
+
+            for ($i=0; $i -lt ($methodParams.Length - $parameterValues.Length); $i++)
+            {
+                $paramValuesAndDefaults.Add([Type]::Missing);
+            } 
+
+            return $method.Invoke($obj, [Reflection.BindingFlags]::OptionalParamBinding, $null, $paramValuesAndDefaults.ToArray(), [Globalization.CultureInfo]::CurrentCulture)
+        }
+    }
+
+    throw "No suitable overload found for $methodName"
+}
+
+#
+# Returns true if the candidate types match are a subset of method parameter types, on a position by position basis
+# 
+function ParamTypesMatch
```

Comment:
What if the method accepts a  base type and you pass in a derived type?
What if there are two methods - one with a base type and one with the derived type? (should invoke based on polymorphic rules).